components/net/http_cache.rs

```diff
@@ -562,6 +580,126 @@ impl HttpCache {
         }
     }
 
+    /// Get the cache entry corresponding to this request
+    pub fn get_entry(&mut self, request: &Request) -> Option<HttpCacheEntry> {
+        if pref!(network.http_cache.disabled) {
+            return None;
+        }
+        if request.method != Method::GET {
+            // Only Get requests are cached.
+            return None;
+        }
+        let entry_key = CacheKey::new(request.clone());
+        let entry = self
+            .entries
+            .entry(entry_key.clone())
+            .or_insert(HttpCacheEntry::new(entry_key));
+        Some(entry.clone())
+    }
+
+    fn invalidate_for_url(&mut self, url: &ServoUrl) {
+        let entry_key = CacheKey::from_servo_url(url);
+        let entry = self
+            .entries
+            .entry(entry_key.clone())
+            .or_insert(HttpCacheEntry::new(entry_key));
+        for cached_resource in entry.resources.write().iter() {
+            cached_resource.write().data.expires = Duration::seconds(0i64);
+        }
+    }
+
+    /// Invalidation.
+    /// <https://tools.ietf.org/html/rfc7234#section-4.4>
+    pub fn invalidate(&mut self, request: &Request, response: &Response) {
+        // TODO(eijebong): Once headers support typed_get, update this to use them
```

Comment:
note: we can remove this TODO(see current usage of `header::LOCATION` and so on)

Reply:
done